functests/1_performance/cpu_management.go

```diff
@@ -349,6 +311,73 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", func() {
 			}
 		})
 	})
+
+	Describe("Verification that IRQ load balance can be disabled per POD", func() {
+		var testpod *corev1.Pod
+
+		BeforeEach(func() {
+			if profile.Spec.GloballyDisableIrqLoadBalancing != nil && *profile.Spec.GloballyDisableIrqLoadBalancing {
+				Skip("IRQ load balance should be enabled (GloballyDisableIrqLoadBalancing=false), skipping test")
+			}
+		})
+
+		AfterEach(func() {
+			deleteTestPod(testpod)
+		})
+
+		It("[test_id:36364] should disable IRQ balance for CPU where POD is running", func() {
+			By("checking default smp affinity is equal to all active CPUs")
+			defaultSmpAffinitySet, err := nodes.GetDefaultSmpAffinitySet(workerRTNode)
+			Expect(err).ToNot(HaveOccurred())
+
+			onlineCPUsSet, err := nodes.GetOnlineCPUsSet(workerRTNode)
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(defaultSmpAffinitySet).To(Equal(onlineCPUsSet), fmt.Sprintf("Default SMP Affinity %s should be equal to all active CPUs %s", defaultSmpAffinitySet, onlineCPUsSet))
+
+			By("Running pod with annotations that disable specific CPU from IRQ balancer")
+			annotations := map[string]string{
+				"irq-load-balancing.crio.io": "disable",
+				"cpu-quota.crio.io":          "disable",
+			}
+			testpod = getTestPodWithAnnotations(annotations)
+
+			err = testclient.Client.Create(context.TODO(), testpod)
+			Expect(err).ToNot(HaveOccurred())
+			err = pods.WaitForCondition(testpod, corev1.PodReady, corev1.ConditionTrue, 10*time.Minute)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Checking that the default smp affinity mask was updated and CPU (where POD is running) isolated")
+			defaultSmpAffinitySet, err = nodes.GetDefaultSmpAffinitySet(workerRTNode)
+			Expect(err).ToNot(HaveOccurred())
+
+			getPsr := []string{"/bin/bash", "-c", "grep Cpus_allowed_list /proc/self/status | awk '{print $2}'"}
+			psr, err := pods.ExecCommandOnPod(testpod, getPsr)
+			Expect(err).ToNot(HaveOccurred())
+			psrSet, err := cpuset.Parse(strings.Trim(string(psr), "\n"))
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(defaultSmpAffinitySet).To(Equal(onlineCPUsSet.Difference(psrSet)), fmt.Sprintf("Default SMP affinity should not contain isolated CPU %s", psr))
+
+			By("Checking that there are no any active IRQ on isolated CPU")
+			// It may takes some time for the system to reschedule active IRQs
+			Eventually(func() string {
+				getActiveIrq := []string{"/bin/bash", "-c", "for n in $(find /proc/irq/ -name smp_affinity_list); do echo $(cat $n); done"}
+				activeIrqList, err := nodes.ExecCommandOnNode(getActiveIrq, workerRTNode)
+				Expect(err).ToNot(HaveOccurred())
+				re := regexp.MustCompile(fmt.Sprintf(`(?m)^%s$`, psrSet))
```

Comment:
Can you please explain this regex?
The psrSet may be  2,3, and the result of activeIrqList may be 0-5,1,4,5
0-5 includes 2,3 would the regex find it?